File: trunk/openrest4net/openrest4net-api/com/googlecode/openrest/Menu.cs
﻿using System;
using System.Collections.Generic;

namespace com.googlecode.openrest
{
    class Menu
    {
        public Menu(IList<Item> items, IList<Tag> tags, IList<Category> categories)
        {
            this.items = items;
            this.tags = tags;
            this.categories = categories;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        internal Menu() { }

        public IList<Item> items = new List<Item>();

        public IList<Tag> tags = new List<Tag>();

        public IList<Category> categories = new List<Category>();
    }
}

File: trunk/openrest4js/openrest4js-helpers/Menu.js
openrest = openrest || {};

openrest.MenuHelper = openrest.MenuHelper || (function() {
	var self = {};

    self.getTagMap = function(params)
    {
        var menu = params.menu;
        var tags = menu.tags || [];
        var tagMap = {};

        for (var i = 0, l = tags.length ; i < l ; i++)
        {
            var tag = tags[i];
            tagMap[tag.id] = tag;
        }

        return tagMap;
    };

	return self;
}());

